src/Processors/Formats/Impl/Parquet/ReadManager.cpp

```diff
@@ -873,7 +873,10 @@ std::tuple<Chunk, BlockMissingValues> ReadManager::read()
             {
                 /// Pump the manual executor.
                 lock.unlock();
-                if (!parser_shared_resources->parsing_runner.runTaskInline())
+                /// Note: the executor can be shared among multiple files, so we may execute someone
+                /// else's task, and someone else may execute our task.
+                /// Hence the thread_pool_was_idle check.
+                if (!parser_shared_resources->parsing_runner.runTaskInline() && thread_pool_was_idle)
```

Comment:
Doesn't it follow from `thread_pool_was_idle` that `runTaskInline()` will always return false? If `active_tasks` was zero (meaning empty queue and no busy threads), there's nothing for `runTaskInline()` to execute.

Reply:
Another file reader could add a task after we checked.